include/deal.II/cgal/utilities.h

```diff
@@ -60,6 +61,17 @@ namespace CGALWrappers
   using ConcurrencyTag = CGAL::Sequential_tag;
 #  endif
 
+  enum class BooleanOperation
+  {
+    corefine     = 1 << 0, ///< Corefine the two surfaces
+    difference12 = 1 << 1, ///< Compute the boolean difference of the first
+                           ///< input argument minus the second
+    difference21 = 1 << 2, ///< Compute the boolean difference of the second
+                           ///< input argument minus the first
+    intersection = 1 << 3, ///< Compute the intersection of the input arguments
+    merge        = 1 << 4, ///< Compute the union of the input arguments
```

Comment:
```suggestion
    union        = 1 << 4, ///< Compute the union of the input arguments
```

Reply:
Also, I would really make sure that the input grids are not changed. This is my proposal:

- corefine: compute the corefinement of the first mesh w.r.t. to the second mesh
- difference: compute the difference between first and second mesh
- interesection: same as it is now
- union: same as it is now

In this way you always write on the output mesh, and never touch the input meshes.

Reply:
I'm fine with your proposal. Also, `union` is a C++ keyword. Maybe I can change the names in `compute_<boolean_operation_name>` ?